tests/test_numpy_array.py

```diff
@@ -595,3 +595,74 @@ def test_round_trip_float():
     arr = np.zeros((), np.float64)
     arr[()] = 37.2
     assert m.round_trip_float(arr) == 37.2
+
+
+# HINT: An easy and robust way (although only manual unfortunately) to check for
```

Comment:
I think it should be possible to do this automatically using weakrefs: https://docs.python.org/3/library/weakref.html#weakref.ref

I'm not sure it's worth changing for this commit.

Reply:
> I think it should be possible to do this automatically using weakrefs:

Yes, but: That only works when making strong assumptions about the reference-counting and garbage collection behavior. In practice, when targeting C Python only, that's usually fine. Fundamentally though such tests are creating tech debt that might get in the way of core interpreter developments in the future. Therefore I generally write tests involving `weakref` and `sys.getrefcount()` only as a last resort.

What would be great to have: a pytest feature, e.g. `@pytest.mark.empirical_leak_check()`, with options like `--empirical_leak_check=quick` and `--empirical_leak_check=thorough`, that run each marked test in a loop for a few seconds and monitor `RES` programmatically, which heuristics to decide how long to try (to deal with noise) and when to flag a potential leak or not. (It will probably be a bit of black magic to get to a useful stable implementation.)